test/utils/IntegrationTest.sol

```diff
@@ -0,0 +1,415 @@
+                linkerCodeHash: forwarderFactory.ERC20LINK_HASH()
+            })
+        );
+
+        poolConfig = IHyperdrive.PoolDeployConfig({
+            baseToken: config.baseToken,
+            linkerFactory: factory.linkerFactory(),
+            linkerCodeHash: factory.linkerCodeHash(),
+            minimumShareReserves: 1e15,
+            minimumTransactionAmount: config.minTransactionAmount,
+            positionDuration: config.positionDuration,
+            checkpointDuration: CHECKPOINT_DURATION,
+            timeStretch: 0,
+            governance: factory.hyperdriveGovernance(),
+            feeCollector: factory.feeCollector(),
+            sweepCollector: factory.sweepCollector(),
+            fees: IHyperdrive.Fees({
+                curve: 0,
+                flat: 0,
+                governanceLP: 0,
+                governanceZombie: 0
+            })
+        });
+    }
+
+    /// @dev A virtual function that exposes important state
+    ///      about the underlying protocol and token. Not all values
+    ///      will be used depending on the instance.
+    /// @param baseSupply Amount of base tokens in circulation.
+    /// @param shareSupply Amount of share tokens in circulation.
+    /// @param sharePrice Price of of one share in terms of base.
+    function getProtocolSharePrice()
+        internal
+        virtual
+        returns (uint256 baseSupply, uint256 shareSupply, uint256 sharePrice);
+
+    /// @dev A virtual function that defines the deployer coordinator
+    ///      contract that will be used to deploy all the instance targets.
+    function deployCoordinator() internal virtual returns (address);
+
+    /// @dev Test to verify a market can be deployed and initialized funded by the
+    ///      base token. Is expected to revert when base deposits are not supported.
+    function test__deployAndInitialize__asBase() external virtual {
+        uint256 aliceBalanceBefore = address(alice).balance;
+
+        // Contribution in terms of base.
+        uint256 contribution = 5_000e18;
+
+        // Get protocol state information used for calculating correct LP shares.
+        (uint256 totalBase, uint256 totalShare, ) = getProtocolSharePrice();
+
+        // Contribution in terms of shares.
+        uint256 contributionShares = contribution.mulDivDown(
+            totalShare,
+            totalBase
+        );
+
+        // Deploy all Hyperdrive targets using deployer coordinator contract.
+        // This function reverts if base deposits are disabled.
+        deployTargets(
+            bytes32(uint256(0xbeefbabe)), // Deployment Id
+            bytes32(uint256(0xdeadfade)), // Deployment Salt
+            contribution, // Contribution
+            true // asBase
+        );
+
+        // If base deposits are enabled we verify some assertions.
+        if (config.enableBaseDeposits && isBaseETH) {
+            // If the base token is ETH we assert the ETH balance is correct.
+            if (isBaseETH) {
```

Comment:
This is implied by the preceding `&&`. Maybe we should remove the `&& isBaseETH`?